src/tests/ftest/dfuse/pil4dfs_fio.py

```diff
@@ -0,0 +1,223 @@
+
+        bws = {}
+        for rw in Pil4dfsFio._FIO_RW_NAMES:
+            self.fio_cmd.update("job", "rw", rw, f"fio --name=job --rw={rw}")
+
+            params = ", ".join(f"{name}={value}" for name, value in self.fio_params.items())
+            self.log.info("Running FIO command: rw=%s, %s", rw, params)
+            self.log.debug(
+                "FIO command: LD_PRELOAD=%s %s", self.fio_cmd.env['LD_PRELOAD'], str(self.fio_cmd))
+            result = self.fio_cmd.run()
+            bws[rw] = self.get_bandwidth(result, rw)
+            self.log.debug("DFuse bandwidths for %s: %s", rw, bws[rw])
+
+        self.assertIsNotNone(self.dfuse)
+        self.log.debug("Stopping DFuse mount point %s", str(self.dfuse))
+        self.stop_dfuse()
+        self.destroy_container()
+
+        return bws
+
+    def run_fio_dfs(self):
+        """Run and return the result of running FIO with DFS ioengine.
+
+        Returns:
+            dict: Read and Write bandwidths of the FIO command.
+
+        """
+        self.create_container()
+
+        self.fio_cmd = FioCommand()
+        self.fio_cmd.get_params(self)
+        self.fio_cmd.update("global", "ioengine", "dfs", "fio --name=global --ioengine='dfs'")
+        self.fio_cmd.update(
+            "global", "numjobs", self.fio_numjobs,
+            f"fio --name=global --numjobs={self.fio_numjobs}")
+        self.fio_cmd.update(
+            "global", "cpus_allowed", self.fio_cpus_allowed,
+            f"fio --name=global --cpus_allowed={self.fio_cpus_allowed}")
+        # NOTE DFS ioengine options must come after the ioengine that defines them is selected.
+        self.fio_cmd.update(
+            "job", "pool", self.pool.uuid,
+            f"fio --name=job --pool={self.pool.uuid}")
+        self.fio_cmd.update(
+            "job", "cont", self.container.uuid,
+            f"fio --name=job --cont={self.container.uuid}")
+        self.fio_cmd.hosts = self.hostlist_clients
+
+        bws = {}
+        for rw in Pil4dfsFio._FIO_RW_NAMES:
+            self.fio_cmd.update("job", "rw", rw, f"fio --name=job --rw={rw}")
+
+            params = ", ".join(f"{name}={value}" for name, value in self.fio_params.items())
+            self.log.info("Running FIO command: rw=%s, %s", rw, params)
+            self.log.debug("FIO command: %s", str(self.fio_cmd))
+            result = self.fio_cmd.run()
+            bws[rw] = self.get_bandwidth(result, rw)
+            self.log.debug("DFS bandwidths for %s: %s", rw, bws[rw])
+
+        self.destroy_container()
+
+        return bws
+
+    def test_pil4dfs_vs_dfs(self):
+        """Jira ID: DAOS-14657.
+
+        Test Description:
+            Run FIO over DFUSE mount point with PIL4DFS interception library
+            Run FIO with DFS ioengine
+            Check bandwidth consistency
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=pil4dfs,dfuse,dfs,fio,thread,fork
```

Comment:
FYI `thread` and `fork` tags aren't used anywhere else. Unless there are plans to use in more tests, I'd recommend removing

Reply:
- [ ] Remove useless tags thread and fork

Reply:
Fixed with commit 4c600bcacb04f835b9ca971db94bf9dadb11c05f